### PR TITLE
chore(Page): convert examples to TypeScript

### LINKED ISSUE #7711: Page: convert examples to TypeScript

### PR DESCRIPTION
**Pull request by Dominik-Petrik**
**What**: Closes #7711

### CODE REVIEW

File: packages/react-core/src/components/Page/examples/GroupSection.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+  const headerToolbar = (
+    <Toolbar id="group-section-toolbar">
+      <ToolbarContent>
+        <ToolbarItem>header-tools</ToolbarItem>
+      </ToolbarContent>
+    </Toolbar>
+  );
+
+  const Header = (
```
Comment (tompsota) [inline packages/react-core/src/components/Page/examples/GroupSection.tsx:42]: I think that the convention is to use camelCase, is that correct @nicolethoen?

```suggestion
  const header = (
```

File: packages/react-core/src/components/Page/examples/CenteredSection.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      </MastheadToggle>
+      <MastheadMain>
+        <MastheadBrand href="https://patternfly.org" target="_blank">
+          Logo
+        </MastheadBrand>
+      </MastheadMain>
+      <MastheadContent>{headerToolbar}</MastheadContent>
+    </Masthead>
+  );
+  const Sidebar = <PageSidebar nav="Navigation" isNavOpen={isNavOpen} id="main-padding-sidebar" />;
```
Comment (tompsota) [inline packages/react-core/src/components/Page/examples/CenteredSection.tsx:55]: I'd add newlines around const declarations (where missing) to make the code easier to read. However I don't have strict opinion on this, what do you think @nicolethoen?

```suggestion

  const Sidebar = <PageSidebar nav="Navigation" isNavOpen={isNavOpen} id="main-padding-sidebar" />;
  
```

**Review by tompsota — CHANGES_REQUESTED** [review]
Hello @Dominik-Petrik, great job with the conversion! :tada: 

I have just a couple of minor comments.

File: packages/react-core/src/components/Page/examples/GroupSection.tsx (hunk 1 above)
Comment (nicolethoen) [reply packages/react-core/src/components/Page/examples/GroupSection.tsx:42]: That's correct - we generally use camelCase for our variable names and PascalCase for component names. Then any text, whether prop description or example titles are all Sentence case.

File: packages/react-core/src/components/Page/examples/CenteredSection.tsx (hunk 2 above)
Comment (nicolethoen) [reply packages/react-core/src/components/Page/examples/CenteredSection.tsx:55]: I don't have a particularly strong opinion here. I think there are definitely times that using line breaks to clearly group related concepts or separate distinct concepts/steps can help with readability. I think in this case, @Dominik-Petrik can make a judgement call if he thinks adding the space is an improvement.

File: packages/react-core/src/components/Page/examples/CenteredSection.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  return (
+    <Page header={Header} sidebar={Sidebar}>
```
Comment (nicolethoen) [inline packages/react-core/src/components/Page/examples/CenteredSection.tsx:57]: The original example actually didn't have a sidebar passed in. Maybe we should keep it that way?

Comment (Dominik-Petrik) [reply packages/react-core/src/components/Page/examples/CenteredSection.tsx:57]: We discussed this via slack and we agreed on maybe it being an oversight so I added the sidebar. But no problem for me to fix it. 😊

Comment (nicolethoen) [reply packages/react-core/src/components/Page/examples/CenteredSection.tsx:57]: nah - sounds good for me! :)

**Comment by Dominik-Petrik** [comment]
I have hopefully resolved all required changes. Let me know if I missed something or if there is anything else that needs to be done.

**Comment by nicolethoen** [comment]
@Dominik-Petrik it looks good, it just needs a rebase! :)

**Review by nicolethoen — APPROVED** [review]

**Review by kmcfaul — APPROVED** [review]

**Pull request merged**